src/shogun/structure/MAPInference.cpp

```diff
@@ -38,6 +39,9 @@ CMAPInference::CMAPInference(CFactorGraph* fg, EMAPInferType inference_method)
 		case GRAPH_CUT:
```

Comment:
BTW, since the factor graph is an input, we could check the graph type, and warn if the inference type is not a best fit.